lib/cadet/accounts/accounts.ex

```diff
@@ -64,6 +64,25 @@ defmodule Cadet.Accounts do
     Repo.get(User, id)
   end
 
+  @doc """
+  Get student with given name and nusnet id or create one
+  """
+  def get_or_create_user(name, role, nusnet_id) do
+    query =
+      from(
+        u in User,
+        where: u.name == ^name and u.role == ^role and u.nusnet_id == ^nusnet_id
+      )
+
+    users = Repo.all(query)
+
+    if length(users) != 0 do
```

Comment:
???

There is a unique index on `nusnet_id` so instead you should be doing
```elixir
User
|> where(name: ^name)
|> where(role: ^role)
|> where(nusnet_id: ^nusnet_id)
|> Repo.one()
|> case do
  nil -> create_user(%{name: name, role: role, nusnet_id: nusnet_id})
  user -> {:ok, user}
end
```

Try to use pattern matching and do not just discard the `:ok` or `:error`